Core/include/Acts/Utilities/DynamicArray.ipp

```diff
@@ -0,0 +1,326 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <exception>
+#include <sstream>
+
+#define THROW_EXCEPTION(MESSAGE)                          \
+  {                                                       \
+    std::stringstream except_str{};                       \
+    except_str << __FILE__ << ":" << __LINE__ << " --- "; \
+    except_str << MESSAGE;                                \
+    throw std::runtime_error(except_str.str());           \
+  }
```

Comment:
this should at least be undef'd afterwards

potentially we want to lift this into some ACTS util or rather drop it or make it switchable and by default off

Reply:
We already have something similar in `ThrowAssert.hpp`.